src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -739,4 +749,56 @@ private RepairRun buildRepairRunFromRow(Row repairRunResult, long id){
         .build(id);
   }
 
+  @Override
+  public boolean takeLeadOnSegment(long segmentId) {
+    Row lwtResult = session.execute(getLeadOnSegmentPrepStmt.bind(segmentId, ReaperApplication.reaperInstanceId, ReaperApplication.getInstanceAddress())).one();
+    if (lwtResult.getBool("[applied]")) {
+      LOG.debug("Took lead on segment {}", segmentId);
+      return true;
+    }
+    
+    // Another instance took the lead on the segment
+    LOG.debug("Could not take lead on segment {}", segmentId);
+    return false;
+  }
+
+  @Override
+  public boolean renewLeadOnSegment(long segmentId) {
+    Row lwtResult = session.execute(renewLeadOnSegmentPrepStmt.bind(ReaperApplication.reaperInstanceId, ReaperApplication.getInstanceAddress(), segmentId, ReaperApplication.reaperInstanceId)).one();
+    if (lwtResult.getBool("[applied]")) {
+      LOG.debug("Renewed lead on segment {}", segmentId);
+      return true;
+    }
+    return false;
+  }
+
+  @Override
+  public void releaseLeadOnSegment(long segmentId) {
+    Row lwtResult = session.execute(releaseLeadOnSegmentPrepStmt.bind(segmentId, ReaperApplication.reaperInstanceId)).one();
```

Comment:
this deserves an assertion check..
if we fail to release the lead on the segment then something has gone wrong and made clear.